File: core/src/main/scala/pl/edu/agh/multiscalemodelling/engine/logic/boudarycondition/FixedBoundaryCondition.java
package pl.edu.agh.multiscalemodelling.engine.logic.boudarycondition;

import pl.edu.agh.multiscalemodelling.engine.logic.Point;

public class FixedBoundaryCondition extends BoundaryCondition {

    @Override
    public Point getPosition(Point position, Point size) {
        return position;
    }

    @Override
    public boolean skip(Point position, Point size) {
        if (position.x < 0 || position.y < 0 || position.x >= size.x || position.y >= size.y) return true;
        return false;
    }
}

File: core/src/main/scala/pl/edu/agh/multiscalemodelling/engine/logic/boudarycondition/FixedBoundaryCondition.scala
package pl.edu.agh.multiscalemodelling.engine.logic.boudarycondition

import pl.edu.agh.multiscalemodelling.engine.logic.Point

class FixedBoundaryCondition extends BoundaryCondition {
  override def getPosition(position: Point, size: Point): Point = position

  override def skip(position: Point, size: Point): Boolean = {
    if (position.x < 0 || position.y < 0 || position.x >= size.x || position.y >= size.y) return true
    false
  }
}